src/front/wgsl.rs

```diff
@@ -1300,7 +1300,7 @@ impl Parser {
                 lexer.expect(Token::Operation('='))?;
                 let inner = self.parse_const_expression(lexer, &mut module.types, &mut module.constants)?;
                 lexer.expect(Token::Separator(';'))?;
-                let const_handle = module.constants.append(crate::Constant {
+                let const_handle = module.constants.fetch_or_append(crate::Constant {
```

Comment:
We are walking over the top level declarations in WGSL here. Why would there be redundant constants or variables?

Reply:
Yes you are right about this one. As the names always have to be unique variables, read [here](https://gpuweb.github.io/gpuweb/wgsl.html#validation). However, there is not a mention about constants not being able to be duplicated, but I guess that should also be the case. I am removing this.